01_fib/mohitnag/main.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fib(7)
+}
+
+func fibonacci(n int) []int {
+	slice := []int{}
+	if n < 1 {
+		return []int{}
+	}
+	if n == 1 {
+		return []int{1}
+	}
+	slice = append(slice, 1, 1)
```

Comment:
you already know how big slice has to be, so it's nicer to pre-allocate the space.
You could use something like:
```go
	if n < 1 {
		return nil
	}
	s := make([]int, n+1)
	s[0], s[1] = 0, 1
	for i := 2; i <= n; i++ {
		s[i] = s[i-1] + s[i-2]
	}
	return s[1:]
```

Reply:
Done